### PR TITLE
ODC-7377,ODC-7378: Add Revisions, Routes and Pods tab for service details page

### PR DESCRIPTION
**Pull request by lokanandaprabhu**
**Story:**
https://issues.redhat.com/browse/ODC-7377
https://issues.redhat.com/browse/ODC-7378

**Description**:

In the developer perspective, in the left side navigation menu, added Functions tab inside Resources section, which list down all the serverless functions created for the specific namepsace and on click of function it opens the Service details tab

Added Revisions, Routes and Pods tab inside Service details page for the developer perspective

**Screenshots:**

https://github.com/openshift/console/assets/102503482/08123f94-3d0a-4daf-824e-f28ec40a14e2

https://github.com/openshift/console/assets/102503482/07dbf201-cb7b-4d83-85f6-6178dbc15229

**Tests:**
Screenshot 2023-09-08 at 1 02 59 PM

### CODE REVIEW

**Review by jerolimov — CHANGES_REQUESTED** [review]
@lokanandaprabhu This PR includes the changes from #13138, is that needed?

I will take a look anyway, focusing only on the new tabs. But when it doesn't depend on the other PR I would recommend dropping the first commit.

If needed, that's okay, but then we should merge the other PR first.

**Comment by lokanandaprabhu** [comment]
@jerolimov this change is added on top of https://github.com/openshift/console/pull/13138, once that PR gets merged I will rebase this. That PR to be checked 1st.

**Comment by lokanandaprabhu** [comment]
Propage labels from the epic:

/label docs-approved
/label px-approved

**Review by jerolimov — CHANGES_REQUESTED** [review]
Minor findings and everything else look good to me. So I have added approved already, anyone can add lgtm when the comments below are addressed.

/approve

**Comment by lokanandaprabhu** [comment]
Updated the PR and addressed the review comments.

File: frontend/packages/knative-plugin/src/actions/providers.ts
Comment (jerolimov) [inline frontend/packages/knative-plugin/src/actions/providers.ts:105]: When the generated-by annotation isn't set, which means the resource is created with the CLI, it shows now

* Edit Function
* Delete Service

Can you also "copy" the `CommonActionFactory.Delete` action so that it calls the origin action but shows "Delete Function"?

**Review by jerolimov — CHANGES_REQUESTED** [review]
Just one small change request.

And please take a look why the knative CI job is failing: https://prow.ci.openshift.org/pr-history/?org=openshift&repo=console&pr=13174

File: frontend/packages/knative-plugin/src/actions/providers.ts
Comment (lokanandaprabhu) [reply frontend/packages/knative-plugin/src/actions/providers.ts]: Updated the PR. PTAL.

**Review by jerolimov — APPROVED** [review]
Works great. Thanks a lot :+1: 

/lgtm

**Pull request merged**
